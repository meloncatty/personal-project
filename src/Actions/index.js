import {apiKey} from '../apiKey.js'

export const resultsSuccess = (results) => ({
  type: 'QUERY_RESULTS_SUCCESS',
  results
})

export const resultsAreLoading = (bool) => ({
  type: 'RESULTS_ARE_LOADING',
  resultsLoading: bool
})

export const resultsHaveErrored = (bool) => ({
  type: 'RESULTS_HAVE_ERRORED',
  resultsErrored: bool
})

export const fullArticleLoading = (bool) => ({
  type: 'FULL_ARTICLE_LOADING',
  articleLoading: bool
})

export const fullArticleErrored = (bool) => ({
  type: 'FULL_ARTICLE_ERRORED',
  articleErrored: bool
})

export const fullArticleSuccess = (result) => ({
  type: 'FULL_ARTICLE_SUCCESS',
  result
})

export const userAuthentication = (userData) => ({
  type: 'USER_AUTHENTICATION',
  userAuth: userData
})

export const isUserSignedIn = (bool) => ({
  type: 'IS_USER_SIGNED_IN',
  isUserSignedIn: bool
})

export const captureUserArticles = (articles) => ({
  type: 'CAPTURE_USER_ARTICLES',
  userArticles: articles
})

export const fetchArticles = (query) => {
  return async (dispatch) => {
    try {
      dispatch(resultsSuccess([]))
      dispatch(resultsHaveErrored(false))
      dispatch(resultsAreLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/search/${query}?page=1&pageSize=10&metadata=true&fulltext=false&citations=true&similar=false&duplicate=false&urls=true&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(resultsAreLoading(false))
      const articles = await response.json()
      dispatch(resultsSuccess(articles.data))
    } catch (error) {
      dispatch(resultsHaveErrored(true))
      dispatch(resultsAreLoading(false))
    }
  }
}

export const fetchFullText = (id) => {
  return async (dispatch) => {
    try {
      dispatch(fullArticleLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/get/${id}?metadata=true&fulltext=true&citations=false&similar=false&duplicate=false&urls=false&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(fullArticleLoading(false))
      const article = await response.json()
      dispatch(fullArticleSuccess(article.data))
    } catch (error) {
      dispatch(fullArticleErrored(true))
      dispatch(fullArticleLoading(false))
    }
  }
}
