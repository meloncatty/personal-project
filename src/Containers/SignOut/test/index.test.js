import React from 'react'
import {shallow} from 'enzyme'
import {SignOutButton, mapDispatchToProps} from '../'
import {isUserSignedIn} from '../../../Actions'

describe('SignOut', () => {
  let signOutButton

  beforeEach(() => {
    signOutButton = shallow(<SignOutButton isUserSignedIn={jest.fn()} userAuthentication={jest.fn()}/>)
  })
  it('should match snapshot', async () => {
    expect(signOutButton).toMatchSnapshot()
  })

  it('should unauthenticate user on sign out', () => {
    signOutButton.find('.sign-out').simulate('click')

    expect(signOutButton.instance().props.userAuthentication).toHaveBeenCalledWith(false)
  })

  it('should update prop isUserSignedIn on sign out', () => {
    signOutButton.find('.sign-out').simulate('click')

    expect(signOutButton.instance().props.isUserSignedIn).toHaveBeenCalledWith(false)
  })

  describe('mapDispatchToProps', () => {
    it('should return false when user signs out', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = isUserSignedIn(false)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.isUserSignedIn(false)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
