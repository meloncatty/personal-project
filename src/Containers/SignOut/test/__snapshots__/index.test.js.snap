// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOutButton
    isUserSignedIn={[Function]}
    userAuthentication={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="sign-out"
        onClick={[Function]}
        replace={false}
        to="/"
>
        Sign Out
</Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Sign Out",
        "className": "sign-out",
        "onClick": [Function],
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": "Sign Out",
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="sign-out"
          onClick={[Function]}
          replace={false}
          to="/"
>
          Sign Out
</Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Sign Out",
          "className": "sign-out",
          "onClick": [Function],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
